helix_fhir_client_sdk/dictionary_parser.py

```diff
@@ -28,32 +28,30 @@ def get_nested_property(
     ) -> Union[List[Dict[str, Any]], Dict[str, Any], str, None]:
         if parent is None:
             return None
+
         parts: List[str] = path.split(".")
-        result: Union[List[Dict[str, Any]], Dict[str, Any]] = parent
+        result: Union[List[Dict[str, Any]], Dict[str, Any], None] = parent
+
         for part in parts:
             if result is None:
                 return None
-            if part.endswith("[x]"):  # list
-                clean_part = part[:-3]
-                if isinstance(result, list):
-                    new_result: List[Dict[str, Any]] = []
-                    for result_entry in result:
-                        value = result_entry.get(clean_part)
-                        if value is not None:
-                            new_result.append(value)
-                    result = new_result
-                else:
-                    result = result.get(clean_part, None)
+            elif part.endswith("[x]") and isinstance(result, list):
+                result = [
+                    value
+                    for result_entry in result
+                    if (value := result_entry.get(part[:-3]))
+                ]
+            elif part.endswith("[x]") and isinstance(result, dict):
+                result = result.get(part[:-3], None)
+            elif isinstance(result, list):
+                result = DictionaryParser.flatten(
+                    [
+                        DictionaryParser.get_nested_property(parent=r, path=part)
+                        for r in result
+                        if r is not None
+                    ]
+                )
             else:
-                if isinstance(result, list):
-                    result = DictionaryParser.flatten(
-                        [
-                            DictionaryParser.get_nested_property(parent=r, path=part)
-                            for r in result
-                            if r is not None
-                        ]
-                    )
-                else:
-                    result = result.get(part, None)
+                result = result.get(part, None)
```

Comment:
Removed the nested if to make it a little more clear what is happening.  was able to remove a for loop in favor of a list comprehension as well.